browser_use/browser/context.py

```diff
@@ -219,6 +219,9 @@ def __init__(
 		state: BrowserContextState | None = None,
 	):
 		self.context_id = str(uuid.uuid4())
+		if not browser.config.headless and config.no_viewport:
+			# keep the window_width and window_height to default values
+			pass  # TODO
```

Comment:
when no_viewport == True and headless == False , then the context state doesn't match with actual browser viewport

```python
assert context.config.window_height == default_viewport_size['height']
assert context.config.window_width == default_viewport_size['width']
```

Reply:
below line of code sets the window.outerWidth,  window.outerHeight of the browser

https://github.com/browser-use/browser-use/blob/942935f0be767562ad08abfa3ffb91168c60d36c/browser_use/browser/browser.py#L322

Reply:
imo window_size and window_height should refer to the inner size of the viewport, and headful browsers should never use viewport